app/javascript/gobierto_dashboards/webapp/containers/contracts/Summary.vue

```diff
@@ -176,12 +228,168 @@ export default {
+      dataLineChart: undefined,
+      valuesForLineChart: undefined,
+      valuesForCircleChart: undefined,
+      labelRootKey: I18n.t('gobierto_dashboards.dashboards.contracts.assignees'),
     }
   },
-  created() {
+  watch: {
+    dashboardsData(newValue, oldValue) {
+      if (newValue !== oldValue) {
+        this.updateDataBeesWarm(newValue)
+      }
+    }
+  },
+  async created() {
     this.columns = assigneesColumns;
+    this.dataBeesWarmFilter = JSON.parse(JSON.stringify(this.dashboardsData));
+
+    const { data: { data: dataLineChart } } = await getQueryData(this.queryLineChart)
+    this.transformDataContractsLine(dataLineChart)
   },
   methods: {
+    updateDataBeesWarm(data){
+      const dataBeesWarm = JSON.parse(JSON.stringify(data));
+      this.dataBeesWarmFilter = dataBeesWarm
+    },
+    transformDataContractsLine(data) {
+      data.forEach(d => {
+        d.final_amount_no_taxes = +d.final_amount_no_taxes
+        d.initial_amount_no_taxes = +d.initial_amount_no_taxes
+        //Calculate percentage median between initial amount and final amount to obtain the difference.
+        d.year = new Date(d.start_date).getFullYear()
+
+        if (d.status === "Formalizado" || d.status === "Adjudicado") {
+          d.formalized = 1
+        } else {
+          d.formalized = 0
+        }
+
+        if (d.status === "Desierto") {
+          d.anulled = 1
+        } else {
+          d.anulled = 0
+        }
+
+        if (d.final_amount_no_taxes === 0) {
+          d.formalized = 0
+          d.anulled = 1
+        }
+      })
+
+      //JS convert null years to 1970
+      let NEXT_YEAR = new Date()
+      let GET_NEXT_YEAR = NEXT_YEAR.getFullYear() + 1
+      data = data.filter(({ year }) => year !== 1970 && year !== GET_NEXT_YEAR)
+
+      let dataFormalizeContracts = data.filter(({ formalized }) => formalized === 1)
+      dataFormalizeContracts.forEach(d => {
+        d.percentage_total = Math.abs(((d.final_amount_no_taxes - d.initial_amount_no_taxes) / d.initial_amount_no_taxes) * 100)
+      })
+      //We need to group and sum by year and value
+      const finalAmountTotal = sumDataByGroupKey(data, 'year', 'final_amount_no_taxes')
+      const formalizedTotal = sumDataByGroupKey(data, 'year', 'formalized')
+      const anulledTotal = sumDataByGroupKey(data, 'year', 'anulled')
+      const initialAmountTotal = sumDataByGroupKey(data, 'year', 'initial_amount_no_taxes')
+      const percentageTotal = sumDataByGroupKey(dataFormalizeContracts, 'year', 'percentage_total')
+
+      //Create a new object with the sum of the properties
+      let dataContractsLine = finalAmountTotal.map((item, i) => Object.assign({}, item, initialAmountTotal[i], percentageTotal[i], formalizedTotal[i], anulledTotal[i]));
+
+      dataContractsLine.forEach(d => {
+        //Get the total of contracts
+        d.total_contracts = (d.anulled + d.formalized)
+
+        d.percentage_year = (d.percentage_total / d.total_contracts)
```

Comment:
Same division by zero. Easy avoid:
```suggestion
        d.percentage_year = d.total_contracts ? (d.percentage_total / d.total_contracts) : 0
```